html/pfappserver/root/static.alt/src/store/base/file.js

```diff
@@ -0,0 +1,210 @@
+                const decoded = new TextDecoder(state.file.encoding).decode(result)
+                resolve(decoded)
+              }).catch(err => {
+                reject(err)
+              })
+            } else {
+              resolve(undefined)
+            }
+          })
+        })
+      },
+      readSlice: ({ commit, state }, { start, end }) => {
+        return new Promise((resolve, reject) => {
+          const reader = new FileReader()
+          reader.onerror = (event) => {
+            const { target: { error } = {} } = event
+            commit('SET_ERROR', error)
+            reader.abort()
+            reject(error)
+          }
+          reader.onload = (event) => {
+            const { target: { result } = {} } = event
+            resolve(new Uint8Array(result))
+          }
+          start = Math.min(start, state.file.size)
+          end = Math.min(end, state.file.size)
+          const slice = state.blob.slice(start, end, state.file.type)
+          reader.readAsArrayBuffer(slice)
+        })
+      },
+      buildOffset: ({ commit, state, dispatch }, lineIndex) => {
+        return new Promise((resolve, reject) => {
+          const scan = async (index, start) => {
+            const end = start + state.chunkSize
+            let match = 0
+            if (index <= lineNumber && state.offsets[index - 1] !== null) {
+              await dispatch('readSlice', { start, end }).then(result => {
+                let offset
+                for (let c = 0; c <= result.length; c++) {
+                  offset = start + c
+                  if (offset === state.file.size) { // EOF
+                    commit('SET_OFFSET', { index, offset: null })
+                    resolve()
+                    return
+                  } else if (result[c] === state.newLine[match]) {
+                    match++
+                    if (match === state.newLine.length) { // EOL
+                      match = 0
+                      commit('SET_OFFSET', { index: index++, offset: offset + 1 })
+                      if (index > lineNumber) break
+                    }
+                  } else {
+                    match = 0
+                  }
+                }
+                start += state.chunkSize
+                scan(index, start) // recurse
+              }).catch(error => {
+                reject(error)
+              })
+            } else {
+              resolve()
+            }
+          }
+          const lineNumber = lineIndex + 1
+          let index = state.offsets.length
+          if (index <= lineNumber && state.offsets[index - 1] !== null) {
+            let start = state.offsets[index - 1] // start at last known offset
+            scan(index, start)
+          } else {
+            resolve()
+          }
+        })
+      }
+    }
+
+    const mutations = {
+      SET_PERCENT: (state, percent) => {
+        state.percent = percent
+      },
```

Comment:
This mutation is not used.

Reply:
This was included in case a component wishes to use this and show an upload progress w/ `read-as-text` on very large files.